### PR TITLE
Move Storefront admin notice content to the Storefront Appearance page.

### LINKED ISSUE #1539: Move Storefront setup notice to an inbox message
Storefront registers a big notice when installed.
image

We need to move that notice an inbox message.
We also need to omit the 'add example product' because that step should be covered by the task list.

### PR DESCRIPTION
**Pull request by budzanowski**
Fixes #1539

We were requested to move the Storefront admin notice to the inbox messages.
Scanning for `admin_notice` has shown that Storefront uses only one notices. This is the main welcome "giant" notice:
image

I have decided to preserve the old `admin_notices` functionality for setups that use WooCommerce versions < 4.6.0.
This version was chosen because from this version the new WooCommerce home screen that was introduced in WooCoomerce 4.3.0 has become the default selection.

### Screenshots

From WooCommerce->Home:

image

From other WooCommerce submenus:

image

### How to test the changes in this Pull Request:

1. Use WooCommerce 4.6.0+
2. On a fresh site ( or with wiped storefront options ) install and activate this Storefront Branch
3. Observe WooCommerce->Home or other WooCommerce submenus inbox area.
4. Click on the `Let's go!` button it should take you to the customizer like previously. It should behave like if the `Apply storefront homepage template` and `Add example products` would be unchecked/

This implementation is missing the homepage template and the example products feature. We may be in the scenario where we don't actually need this because wc-admin setup list task `Personalize my store` is doing that so doubling this feature would be counterproductive. The discussion is taking place here: pbIJXs-s6-p2
The inbox messages should not be used the way we used our admin_notice. We can't add a form that would host our checkboxes. One of the options I see is to use 3 inbox messages that would exist as separate actions. This sounds reasonable because the inbox is designed to accommodate multiple messages. Although I think it is OK this approach needs design guiding. 
I have tested how it would look for two:
image
The Add products button is not connected, it is just a mockup, but it shows how this approach would look.

If using multiple messages is not an option we would need to take a look at the tasks list direction.

@haszari thoughts?

@LevinMedia If we would go one message or multiple messages options, could you please help with copy and overall design. There is little flexibility with Inbox Messages so probably copy only, or we can have a banner :P 

### Changelog

Move Storefront welcome notice to the WooCommerce Admin inbox messages.

### CODE REVIEW

**Comment by haszari** [comment]
Note - I haven't reviewed, just some initial thoughts.

Would it be better to wait and expose this as an onboarding task (rather than inbox notice)? Since this triggers a setup/onboarding step.

The previous notice had checkboxes to give the user control over what happened - are these options still available?

**Comment by garymurray** [comment]
As per the conversation here: pca54o-Fy-p2 

- Let's move the current wp-admin notice that Storefront generates to `Appearance > Storefront` 
- Add an Inbox notice that sends merchants to that screen to configure Storefront

**Comment by budzanowski** [comment]
Looks like the core work is done.
TODO:
- [x] Check if everything works ( fresh site ).
- [x] Update information in the PR.
- [x] Update screenshots.
- [ ] Check the documentation. Does it need any updates?
- [x] Ask design for input.
- [x] Inbox message copy.
- [x] Self code review.
- [x] Review.
- [ ] Merge to trunk or release/3.0.

**Comment by budzanowski** [comment]
image

**Comment by budzanowski** [comment]
@garymurray After moving the notice I realized that the page looked very awkward. Cluttered, not very friendly and welcoming. I have decided to get rid of the notice and incorporate its content directly into the page. It looks much better this way. You can see how it looks in the screenshot above.

All the changes made are just for WC 4.6+ all the other versions display everything as they used to.

**Comment by garymurray** [comment]
The visual changes LGTM thanks @budzanowski

**Comment by budzanowski** [comment]
## PR Updated

All changes are applied only to WC 4.6+. This version was chosen because this was the moment when the `wc-admin` homepage has become the default WooCommerce Home screen.

The `admin_notice` is no longer used. Instead, a Storefront welcome message is added to the Inbox:

<p align="center">
  
</p>

Clicking `Let's go` takes the user to the `Appearance` -> `Storefront` page.  This page now has the content of the admin notice incorporated into the page. Othe original text is not gone! It will be visible after setup or skip action:

<p align="center">
  
</p>

That section behaves exactly as the old `admin_notice` did. The same code is handling them so behaviour wise there should be 0 differences. The notice `X` dismiss button was replaced with the `Skip` link. `Skip` is supported by a very simple hide/fadeIn animation to make everything feel smooth:

<p align="center">
  
</p>

@LevinMedia design OK?
@haszari please review :)

**Review by haszari — COMMENTED** [review]
I hit a couple of issues with this, mostly minor/fixable but one fatal! Note I haven't tested all paths or fully reviewed code yet.

Issues:

1. Fatal error - `WooCommerce\Admin\Notes\Note` class not found
2. Inbox notice is buried at bottom of screen / not in onboarding task list

### Fatal error - `WooCommerce\Admin\Notes\Note` class not found
I hit this fatal when activating Storefront (freshly built zip from this branch) on a Jurassic Ninja site. The only plugins active are WooCommerce 4.7.1 (release) and Jetpack . (Full Woo status report below with versions etc.)

Here's the full error:

`Fatal error: Uncaught Error: Class 'Automattic\WooCommerce\Admin\Notes\Note' not found in /srv/users/userf5d002c0/apps/userf5d002c0/public/wp-content/themes/storefront/inc/nux/class-storefront-nux-admin-inbox-messages-customize.php:32 Stack trace: #0 /srv/users/userf5d002c0/apps/userf5d002c0/public/wp-content/plugins/woocommerce/packages/woocommerce-admin/src/Notes/NoteTraits.php(74): Storefront_NUX_Admin_Inbox_Messages_Customize::get_note() #1 /srv/users/userf5d002c0/apps/userf5d002c0/public/wp-content/themes/storefront/inc/nux/class-storefront-nux-admin.php(194): Storefront_NUX_Admin_Inbox_Messages_Customize::possibly_add_note() #2 /srv/users/userf5d002c0/apps/userf5d002c0/public/wp-includes/class-wp-hook.php(287): Storefront_NUX_Admin->admin_inbox_messages('') #3 /srv/users/userf5d002c0/apps/userf5d002c0/public/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters(NULL, Array) #4 /srv/users/userf5d002c0/apps/userf5d002c0/public/wp-includes/plugin.php(478): WP_Hook->do_action(Array) #5 /srv/users/userf5d002c0/apps/ in /srv/users/userf5d002c0/apps/userf5d002c0/public/wp-content/themes/storefront/inc/nux/class-storefront-nux-admin-inbox-messages-customize.php on line 32`

<details>

```
### WordPress Environment ###

WordPress address (URL): https://hon-ladybird.jurassic.ninja
Site address (URL): https://hon-ladybird.jurassic.ninja
WC Version: 4.7.1
REST API Version: ✔ 4.7.1
WC Blocks Version: ✔ 3.6.0
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.6.3
Log Directory Writable: ✔
WP Version: 5.5.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.46 (Unix) OpenSSL/1.0.2g
PHP Version: 7.4.12
PHP Post Max Size: 1 GB
PHP Time Limit: 30
PHP Max Input Vars: 5000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
MySQL Version: 5.7.32-0ubuntu0.16.04.1-log
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.7.1
WC Database Prefix: wp_
Total Database Size: 3.70MB
Database Data Size: 2.32MB
Database Index Size: 1.38MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 1.48MB + Index: 0.03MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 6
post: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (3) ###

Companion Plugin: by Osk – 1.18
Jetpack by WordPress.com: by Automattic – 9.2
WooCommerce: by Automattic – 4.7.1

### Inactive Plugins (2) ###

Akismet Anti-Spam: by Automattic – 4.1.7
Hello Dolly: by Matt Mullenweg – 1.7.2

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty
Version: 1.5
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 1
Oldest: 2020-12-08 02:38:28 +0000
Newest: 2020-12-08 02:38:28 +0000
```

</details>

Perhaps there's another version check we need before we use the inbox APIs? 

Screen Shot 2020-12-08 at 3 54 19 PM

### Inbox notice is buried at bottom of screen / not in onboarding task list
This is more design feedback, not necessarily a blocker. Keen to hear thoughts from @LevinMedia. When testing locally (with latest Woo core `master`, so no fatal), I get the new onboarding notice and the new Storefront admin screen.

The notice itself is somewhat buried and de-emphasised in the admin home screen. I think this is because the user is still in onboarding (the task list, top or top-right). Note I was testing on a MacBook 13" screen; on bigger screen the notice slightly less buried.

Storefront welcome notice - buried at bottom, and grey:

Screen Shot 2020-12-08 at 3 18 41 PM

Task list is highlighted (bright white) by default:

Screen Shot 2020-12-08 at 3 18 36 PM

Not sure how much of an issue this is – I see there is discussion of an inbox notice OR a task list item on the P2 (pbIJXs-s6-p2). There's definitely a bigger problem here, that the Storefront onboarding is complex and overlaps with Woo Core onboarding. This has been [discussed before](https://github.com/woocommerce/storefront/issues/1432), with the outcome that we just leave as is for now. So this is a tricky area, hopefully there's a way we can safely improve the UX and lean in to core onboarding more.

**Comment by budzanowski** [comment]
@haszari thank you for testing.

Oh no 😀. Thanks for testing. Yes, there is a bug on 4.7.1. I am sorry for that. I was developing on 4.8.0-dev. And I also did the testing on the latest dev. I need to update my testing notes to not forget about older versions. It turns out that woocommerce-admin has recently done a name refactor for the Notes classes:

https://github.com/woocommerce/woocommerce-admin/pull/5142

I did a quick update to the code and changed the required version to 4.8.0, which makes sense either way given how we target the Storefront releases. We also avoid deprecation notices this way.

The notice is darker only when it was viewed ( I don't know what exact logic is used here ) but initially it visible with a white background like the first inbox message you see in the list. When you are using the inbox messages widget ( in other than home pages under woocommerce ) the message is initially on the top of the list.

I think that this is good, but I don't want to push is unless there is a consensus in the team that we want to do this now.
And I don't think that we agree on how to proceed - which is fine, we just need to make a decision.

**Comment by nerrad** [comment]
Just noting that from an implementation perspective (without a code review or testing) - I'm okay with including this in a release as is provided it tests well. With this going into another Storefront RC release first, we'd still have the ability to revert if we decide to remove before final release.

**Comment by LevinMedia** [comment]
@budzanowski @haszari  - I think the design and copy work for now. Can you add a space between "Storefront" and the emoji, please?

File: inc/nux/class-storefront-nux-admin-inbox-messages-customize.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+	/**
+	 * Get the note.
+	 *
+	 * @return Note
+	 */
+	public static function get_note() {
+		$note = new Note();
+		$note->set_title( __( 'Design your store with Storefront 🎨', 'storefront' ) );
+		$note->set_content( __( 'Visit the Storefront settings page to start setup and customization of your Shop.', 'storefront' ) );
```
Comment (haszari) [inline inc/nux/class-storefront-nux-admin-inbox-messages-customize.php]: Nit - should `Shop` be capitalised?

File: inc/nux/class-storefront-nux-admin-inbox-messages-customize.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		$note->set_type( Note::E_WC_ADMIN_NOTE_INFORMATIONAL );
+		$note->set_name( self::NOTE_NAME );
+		$note->set_content_data( (object) array() );
```
Comment (haszari) [inline inc/nux/class-storefront-nux-admin-inbox-messages-customize.php:37]: Is this needed? Seems strange to add an empty array.

File: inc/nux/class-storefront-nux-admin.php [hunk 3]
```diff
@@ -150,6 +178,21 @@ public function dismiss_nux() {
...
+		 * @since 3.0.0
+		 */
+		public function admin_inbox_messages() {
+			// The setup already has happened. No inbox message needed.
+			if ( true === (bool) get_option( 'storefront_nux_dismissed' ) ) {
+				return;
+			}
+			// Storefront settings page link and welcome message.
+			require 'class-storefront-nux-admin-inbox-messages-customize.php';
+			Storefront_NUX_Admin_Inbox_Messages_Customize::possibly_add_note();
```
Comment (haszari) [inline inc/nux/class-storefront-nux-admin.php:193]: Where's `possibly_add_note()` defined? I see `get_note()`, this is working, so I must be confused.

**Review by haszari — APPROVED** [review]
Thanks for the recent fixes and improvements @budzanowski !

I've tested this with Woo 4.7.1 and Woo 4.8 and found no problems. I tested a range of scenarios - dismissing the inbox notice, skipping, triggering onboarding with different existing state and options (products, homepage data/checkboxes).

Having the onboarding live in `Themes > Storefront` is a nice enhancement - it's easy to find, and not shown on every admin page as an admin notice. This might decrease the visibility/usage a bit but I think it's a much better experience (than having a big notice everywhere!). Also the behaviour when the inbox notice is dismissed is nice - the onboarding is still available in `Themes > Storefront` page 🙌 

One quirk I noticed is that when there's an existing home page, the `Apply the Storefront homepage template` onboarding option doesn't work very well. This is not new, just a reminder that when we go looking we may find quirks in our complex onboarding.

👍 🚢 

Good to merge and ship - thanks for your persistence improving this!

File: inc/nux/class-storefront-nux-admin-inbox-messages-customize.php (hunk 1 above)
Comment (budzanowski) [reply inc/nux/class-storefront-nux-admin-inbox-messages-customize.php]: nope. Fixed.

File: inc/nux/class-storefront-nux-admin-inbox-messages-customize.php (hunk 2 above)
Comment (budzanowski) [reply inc/nux/class-storefront-nux-admin-inbox-messages-customize.php:37]: It is needed. I have seen a comment about making this default when nothing is provided but for now, we need this.

File: inc/nux/class-storefront-nux-admin.php (hunk 3 above)
Comment (budzanowski) [reply inc/nux/class-storefront-nux-admin.php:193]: This is from NoteTraits.

**Pull request merged**